src/Action/Iterator.php

```diff
@@ -66,12 +66,12 @@ public function like($field, $value)
         $this->generator = new \CallbackFilterIterator($this->generator, function ($row) use ($field, $value) {
 
             // skip row. does not have field at all
-            if (!isset($row[$field])) {
+            if (!array_key_exists($field, $row)) {
                 return false;
             }
 
             $clean_value = trim(trim($value), '%');
```

Comment:
The trims are not compatible with the swith below.

Reply:
Why not? Trim gives us clean value which is used below.

Reply:
As the `substr`s in the switch are checking in untrimmed string.

Reply:
Ah, right. Fixed in https://github.com/atk4/data/pull/538